src/Sensors/IMU/IMU.cpp

```diff
@@ -54,14 +66,98 @@ namespace mmfs
         sprintf(data, "Initial Magnetic Field (uT): %.2f,%.2f,%.2f\n", initialMagField.x(), initialMagField.y(), initialMagField.z());
         return data;
     }
-    void IMU::update()
+
+    void IMU::quaternionBasedComplimentaryFilter(double dt)
```

Comment:
What calls this function?

Reply:
This function can be thought of as util functions native to IMUs and get called during the hardware read() step if that hardware needs it to find orientation.